### ISSUE TITLE
Handling further error cases for reading the history file

### ISSUE THREAD

**Issue by LostKobrakai** [mattludwigs/mobius#28]
Mobius crashes if the history file loaded by `Mobius.Scraper.load_history/2` is corrupted or otherwise not readable, but existing. This probably should be handled more gracefully.

**Comment by mdwaud** [comment]
I've been super happy with Mobius so far, but am seeing a related issue:

```Elixir
    ** (EXIT) shutdown: failed to start child: Mobius
        ** (EXIT) shutdown: failed to start child: Mobius.Scraper
            ** (EXIT) an exception was raised:
                ** (MatchError) no match of right hand side value: {:error, :unsupported_version}
                    (mobius 0.3.2) lib/mobius/scraper.ex:80: Mobius.Scraper.load_history/2
                    (mobius 0.3.2) lib/mobius/scraper.ex:69: Mobius.Scraper.make_history/2
                    (mobius 0.3.2) lib/mobius/scraper.ex:54: Mobius.Scraper.init/1
                    (stdlib 3.13) gen_server.erl:417: :gen_server.init_it/2
                    (stdlib 3.13) gen_server.erl:385: :gen_server.init_it/6
                    (stdlib 3.13) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

I could see two approaches to dealing with these issues:

- approach 1: delete the old metrics file
- approach 2: backup the old metrics file and create a new one

**Comment by mdwaud** [comment]
I took approach 3: ignore the old, unreadable metrics file.

**Comment by mattludwigs** [comment]
@mdwaud and @LostKobrakai thanks for reporting this issue!

@mdwaud I like the PR, I am just thinking about the implications.

Once the new history is written to the filesystem, the bad file will be overwritten. This would lead to historical information being lost. This might not matter as there might not be a way to recover the old data.

**Comment by mattludwigs** [comment]
Closed by #29, thank you @mdwaud!
